lightning/src/routing/router.rs

```diff
@@ -3400,6 +3538,361 @@
 			assert_eq!(total_amount_paid_msat, 90_000);
 		}
 	}
+
+	#[test]
+	fn min_criteria_consistency() {
+		// Test that we don't use an inconsistent metric between updating and walking nodes during
+		// our Dijkstra's pass. In the initial version of MPP, the "best source" for a given node
+		// was updated with a different critera from the heap sorting, resulting in loops in
+		// calculated paths. We test for that specific case here.
+
+		// We construct a network that looks like this:
+		//
+		//            node2 -1(3)2- node3
```

Comment:
is the bracketed number the value locked up in HTLCs?

Reply:
Its the short_channel_id. I think this is documented or at least more clear in the graphs in some of the earlier tests.